### PR TITLE
FEAT-#4342: Support `from_dataframe` for pandas storage format

### LINKED ISSUE #4342: Support `from_dataframe` for pandas storage format
To fully support the Dataframe exchange protocol for pandas storage format we should add `from_dataframe` to PandasQC and PandasDataframe classes.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4342  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by dchigarev** [comment]
Let's add a simple test here testing `from_dataframe` for pandas on ray execution, something like:
```python
from modin.pandas.utils import from_dataframe
from modin.experimental.core.execution.native.implementations.omnisci_on_native.dataframe import OmnisciOnNativeDataframe

def simple_import():
    omnisci_producer = OmnisciOnNativeDataframe.from_pandas(pd_df)
    # Consider the selected factory is PandasOnRay
    pandas_on_ray_consumer = from_dataframe(omnisci_producer)
    df_equals(omnisci_producer, pandas_on_ray_consumer)

```

File: modin/core/dataframe/pandas/dataframe/dataframe.py [hunk 1]
```diff
@@ -2859,3 +2859,36 @@ def __dataframe__(self, nan_as_null: bool = False, allow_copy: bool = True):
...
+        df : ProtocolDataframe
+            The DataFrame object supporting the dataframe exchange protocol.
+
+        Returns
+        -------
+        PandasDataframe
+            A new Core Modin Dataframe object.
+        """
+        if isinstance(df, cls):
+            return df
```
[the remark this answers is not in the document]
Comment (dchigarev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py:2881]: there's actually the same error in `OmnisciOnNativeDataframe.from_dataframe` implementation:
https://github.com/modin-project/modin/blob/0c1a2129df64cf45bf1ff49c8ed92c510fdb1c82/modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py#L2096-L2097

[the remark this answers is not in the document]
Comment (YarShev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py:2881]: fixed

**Comment by YarShev** [comment]
> Let's add a simple test here for `from_dataframe` for pandas on ray execution, something like:
> 
> ```python
> from modin.pandas.utils import from_dataframe
> from modin.experimental.core.execution.native.implementations.omnisci_on_native.dataframe import OmnisciOnNativeDataframe
> 
> def test_simple_import():
>     omnisci_producer = OmnisciOnNativeDataframe.from_pandas(pd_df)
>     # Consider the selected factory is PandasOnRay
>     pandas_on_ray_consumer = from_dataframe(omnisci_producer)
>     df_equals(omnisci_producer, pandas_on_ray_consumer)
> ```
> 
> We can put it into a new file `modin/test/exchange/dataframe_protocol/pandas/test_protocol.py`

done

**Review by dchigarev — APPROVED** [review]
LGTM assuming CI passes

**Pull request merged**